controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,205 @@
+}
+type InfoStorageCluster struct {
+	Metadata                types.NamespacedName `json:"metadata"`
+	StorageProviderEndpoint string               `json:"storageProviderEndpoint"`
+	CephClusterFSID         string               `json:"cephClusterFSID"`
+}
+
+type OdfInfoData struct {
+	OdfVersion          string             `json:"odfVersion"`
+	OdfDeploymentType   string             `json:"odfDeploymentType"`
+	Clients             []ConnectedClient  `json:"clients"`
+	StorageCluster      InfoStorageCluster `json:"storageCluster"`
+	StorageClusterCount int                `json:"storageClusterCount"`
+	StorageSystemName   string             `json:"storageSystemName"`
+	IsDROptimized       bool               `json:"isDROptimized"`
+}
+
+const (
+	RDROdfInfoKeyName            = "config.yaml"
+	RDROdfDeploymentTypeExternal = "external"
+	RDROdfDeploymentTypeInternal = "internal"
+	RookCephMonSecretName        = "rook-ceph-mon"
+	FsidKey                      = "fsid"
+	OdfOperatorNamePrefix        = "odf-operator"
+	RDROdfInfoConfigMapName      = "odf-info"
+	RDROdfInfoMapKind            = "ConfigMap"
+)
+
+type odfInfoConfig struct{}
+
+// ensureCreated ensures that a ConfigMap resource exists with its Spec in
+// the desired state.
+func (obj *odfInfoConfig) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (reconcile.Result, error) {
+	odfInfo, configErr := getOdfInfoConfig(r, sc)
+	if configErr != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get ODF info config data: %w", configErr)
+	}
+	odfInfoKey := RDROdfInfoKeyName + "." + sc.Name
+	odfInfoConfigOverrideData := map[string]string{
+		odfInfoKey: odfInfo,
+	}
+	odfInfoConfigOverrideCM := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       RDROdfInfoMapKind,
+			APIVersion: sc.APIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      RDROdfInfoConfigMapName,
+			Namespace: sc.Namespace,
+		},
+		Data: odfInfoConfigOverrideData,
+	}
+
+	_, err := ctrl.CreateOrUpdate(context.Background(), r.Client, odfInfoConfigOverrideCM, func() error {
+		if !reflect.DeepEqual(odfInfoConfigOverrideCM.Data, odfInfoConfigOverrideData) {
+			r.Log.Info("updating ODF info config override configmap", RDROdfInfoMapKind, klog.KRef(sc.Namespace, RDROdfInfoConfigMapName))
+			odfInfoConfigOverrideCM.Data = odfInfoConfigOverrideData
+		}
+		return ctrl.SetControllerReference(sc, odfInfoConfigOverrideCM, r.Scheme)
+	})
+	if err != nil {
+		r.Log.Error(err, "failed to create or update ODF info config override", RDROdfInfoMapKind, klog.KRef(sc.Namespace, RDROdfInfoConfigMapName))
+		return reconcile.Result{}, fmt.Errorf("failed to create or update ODF info config override: %w", err)
+	}
+	return reconcile.Result{}, nil
+}
+
+// ensureDeleted is dummy func for the odfInfoConfig
+func (obj *odfInfoConfig) ensureDeleted(_ *StorageClusterReconciler, _ *ocsv1.StorageCluster) (reconcile.Result, error) {
+	return reconcile.Result{}, nil
+}
```

Comment:
If we are going with a distributed update approach, `ensureDeleted` should remove the key for the storagecluster being reconciled from the config map and if it is the last key, delete the entire config map